src/compiler/checker.ts

```diff
@@ -7588,15 +7629,38 @@ namespace ts {
                 links.type = instantiateType(getTypeOfSymbol(lastOrUndefined(context.parameters)), mapper);
             }
         }
+        
+        function createPromiseType(promisedType: Type): Type {

```

Comment:
functions with "create" prefix always return a type. the ones with "get" can return undefined though. consider renaming this function.